src/lib.rs

```diff
@@ -3856,13 +3934,46 @@ impl Connection {
                 }
             },
 
-            frame::Frame::StopSending { stream_id, .. } => {
+            frame::Frame::StopSending {
+                stream_id,
+                error_code,
+            } => {
                 // STOP_SENDING on a receive-only stream is a fatal error.
                 if !stream::is_local(stream_id, self.is_server) &&
                     !stream::is_bidi(stream_id)
                 {
                     return Err(Error::InvalidStreamState);
                 }
+
+                // Get existing stream or create a new one, but if the stream
+                // has already been closed and collected, ignore the frame.
+                //
+                // This can happen if e.g. an ACK frame is lost, and the peer
+                // retransmits another frame before it realizes that the stream
+                // is gone.
+                //
+                // Note that it makes it impossible to check if the frame is
+                // illegal, since we have no state, but since we ignore the
```

Comment:
illegal modulo the previous check

Reply:
Yeah, basically we can't check it's validity against the state of the corresponding stream as that might have already been collected. FWIW this is a copy-pasta, and the same comment is also in the `ResetStream` and `Stream` cases.